### PR TITLE
add k8s native selector

### PR DESCRIPTION
**Pull request by 27149chen**
add k8s native selector

### CODE REVIEW

**Comment by Iceber** [comment]
The feature looks the same as https://github.com/clusterpedia-io/client-go/pull/46, but using the native `labels.Selector` is more generic


@feeltimeQ  I think we can use `Selector` to replace `LabelSelectorOption` and then delete `LabelSelectorOption` directly, what do you think?

**Comment by calvin0327** [comment]
For a variety of very complex scenarios, I also think it's more flexible to just accept a native selector.

**Comment by Iceber** [comment]
@27149chen please rebase

**Comment by feeltimeQ** [comment]
Yes, his will be better

File: tools/builder/options.go [hunk 1]
```diff
@@ -59,6 +60,7 @@ type LabelSelectorOption struct {
 type listOptions struct {
 	options                  metav1.ListOptions
 	labelSeletor             map[string][]string
+	labelSelector            labels.Selector
```
Comment (calvin0327) [inline tools/builder/options.go:63]: the name of var seem to be a conflict with `labelSeletor`.
the `labelSeletor` is a typo 😅

can we redefine these two fields？

Comment (Iceber) [reply tools/builder/options.go:63]: /s/labelSelector/labels ?

Comment (calvin0327) [reply tools/builder/options.go:63]: good idea.

**Comment by 27149chen** [comment]
updated

**Review by Iceber — APPROVED** [review]
LGTM

**Comment by calvin0327** [comment]
LGTM

**Pull request merged**